pkg/controller/pravegacluster/upgrade.go

```diff
@@ -564,6 +581,30 @@ func (r *ReconcilePravegaCluster) transitionToNewSTS(p *pravegav1alpha1.PravegaC
 	}
 	//this is to check if all the new ss pods have comeup before deleteing the old sts
 	if newsts.Status.ReadyReplicas == p.Spec.Pravega.SegmentStoreReplicas {
+		if p.Spec.ExternalAccess.Enabled {
```

Comment:
Again, Encapsulate all the code inside the if block in a method and invoke the method here... something like ...
`if p.Spec.ExternalAccess.Enabled { deleteExternalServices(. . .) } `